README.md

```diff
@@ -70,79 +70,76 @@ We have three aliases for product versions;
     
   * To facilitate merging your changes into upstream, we recommend following
-    our procedure for supporting SDK backward-compatibility if needed. The code
-    for maintaining SDK compatibility lives in
-    [sdkcompat](https://github.com/bazelbuild/intellij/tree/master/sdkcompat) directory.
-    We follow these three techniques for non-trivial incompatible changes.
+    our procedure for supporting SDK backward-compatibility. 
     
-    * **Compat**  
-       Preferred for small changes like a change in the return type of a method. 
-       We add a util-class with only static methods and a private constructor and
-       wrap the changed method by one of the static methods. If the change is small enough,
-       you do not need to create a new util-class and should add a new method in 
-       [BaseSdkCompat class](https://github.com/bazelbuild/intellij/blob/master/sdkcompat/v201/com/google/idea/sdkcompat/general/BaseSdkCompat.java) instead. Example: [pr/2345](https://github.com/bazelbuild/intellij/pull/2345)
-       
-    * **Adapter**  
-       Used when we extend a super class and an overriden method signature changes
-       (e.g. a new parameter is added) or the super class constructor is updated.
-       We create a new class extending the changed super class to override the updated methods appropriately then extend this new class
-       from the plugin code. Example: [pr/2114](https://github.com/bazelbuild/intellij/pull/2114)
-       
-    * **Wrapper**  
-      Created when a new interface is used in a super class constructor. We create
-      a wrapper class that wraps and supplies the old or the new interface based on
-      the SDK version and use this wrapper class in the plugin code.
-      Example: [pr/2166](https://github.com/bazelbuild/intellij/pull/2166)
-
-  * All compat changes should be commented with #api{API_VERSION}, e.g. #api203.
+    * First consider adjusting the plugin code so that it directly works with different IDE versions. 
+      Example strategies for this would be:
+      
+      * Switching to a (possibly newer) IntelliJ platform API which is available in all relevant IDE versions. Example: [pr/2623](https://github.com/bazelbuild/intellij/pull/2623)
+      * Switching to a raw class by removing a generic type parameter which differs across versions. Example: [pr/2631](https://github.com/bazelbuild/intellij/pull/2631)
+    
+    * For non-trivial incompatible changes, the code for maintaining SDK compatibility lives
+      in [sdkcompat](./sdkcompat) and [testing/testcompat](./testing/testcompat) directories. Where `testing/testcompat`
+      holds test-only SDK compatibility changes. Each of the two directories contains a sub-folder per supported IDE version with 
+      version-specific implementations. The outside API of all classes must be the same across versions. 
+      Just the implementation may differ. When introducing a new file in this directory, make sure to duplicate
```

Comment:
Typo: This seems to be a dependent clause again. -> Switch to a comma again: "... across versions, just the implementation ...".